<?xml version="1.0" encoding="utf-8"?>
<!--
 Copyright (c) 2021  Carnegine Mellon University

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
-->

<launch>
  <arg name="map_frame" default=""/>
  <arg name="camera_link_frame" default=""/>
  <arg name="camera_info_topic" default=""/>
  <arg name="image_rect_topic" default=""/>
  <arg name="depth_registered_topic" default=""/>
  <arg name="use_nodelet" default="0"/>
  <arg name="namespace" default="camera"/>
  <arg name="publish_simulator_people" default="false"/>

  <node namespace="$(var namespace)"
	name="detect_darknet_people_cpp" pkg="track_people_cpp" exec="manual_composition" output="log">
    <param from="$(find-pkg-share track_people_py)/params/realsense.yaml" />

    <!--
    <param name="map_frame" value="$(var map_frame)"/>
    <param name="camera_id" value="$(var namespace)"/>
    <param name="camera_link_frame" value="$(var camera_link_frame)"/>
    <param name="camera_info_topic" value="$(var camera_info_topic)"/>
    <param name="image_rect_topic" value="$(var image_rect_topic)"/>
    <param name="depth_registered_topic" value="$(var depth_registered_topic)"/>
-->
    
    <param name="detection_threshold" value="0.25" />
    <param name="minimum_detection_size_threshold" value="50.0" />
    
    <param name="detect_config_file" value="$(find-pkg-share track_people_py)/scripts/yolov4.cfg" />
    <param name="detect_weight_file" value="$(find-pkg-share track_people_py)/scripts/yolov4.weights" />
    <param name="detect_label_file" value="$(find-pkg-share track_people_py)/scripts/coco.names" />
    <!-- cannot load params by include
	 <include file="$(find-pkg-share track_people_py)/launch/detect_common_darknet.launch.xml" />
    -->
  </node>

  <!-- <group ns="$(arg namespace)">
    <rosparam ns="track_people_py" command="load" file="$(find track_people_py)/params/realsense.yaml" />

    <param name="track_people_py/map_frame" value="$(arg map_frame)" if="$(eval map_frame!='')"/>
    <param name="track_people_py/camera_id" value="$(arg namespace)"/>
    <param name="track_people_py/camera_link_frame" value="$(arg camera_link_frame)" if="$(eval camera_link_frame!='')"/>
    <param name="track_people_py/camera_info_topic" value="$(arg camera_info_topic)" if="$(eval camera_info_topic!='')"/>
    <param name="track_people_py/image_rect_topic" value="$(arg image_rect_topic)" if="$(eval image_rect_topic!='')"/>
    <param name="track_people_py/depth_registered_topic" value="$(arg depth_registered_topic)" if="$(eval depth_registered_topic!='')"/>

    <include file="$(find track_people_py)/launch/detect_common_darknet.launch" />

    <node name="detect_darknet_opencv_nodelet" pkg="nodelet" type="nodelet" output="log" if="$(arg use_nodelet)"
	  args="load TrackPeopleCPP/DetectDarknetOpencvNodelet realsense2_camera_manager" >
    </node>

    <node name="detect_darknet_opencv_node" pkg="track_people_cpp" type="detect_darknet_opencv_node" unless="$(arg use_nodelet)"
	  output="log">
    </node>

  </group> -->
</launch>
